servicetest/devicenode/test_devicenode_gateway.py

```diff
@@ -0,0 +1,188 @@
+        "mode": 754
+    }
+]
+
+GW_CONFIG_NEW_DEV_0 = [
+    {
+        "name": "/dev/tty0",
+        "mode": 644
+    }
+]
+
+GW_CONFIG_NEW_DEV_1 = [
+    {
+        "name": "/dev/tty1",
+        "mode": 755
+    }
+]
+
+GW_CONFIG_SAME_DEV_SAME_MODE = [
+    {
+        "name": "/dev/tty0",
+        "mode": 644
+    }
+]
+
+""" Following capabilities are fed to DynamicConfiguration tests to examine the case
+    when new mode is configured for same device.
+"""
+test_cap_0 = Capability("double.activation.same.device.0.cap",
+                        [
+                            {"id": "devicenode", "config": GW_CONFIG_SAME_DEV_0}
+                        ])
+
+test_cap_1 = Capability("double.activation.same.device.1.cap",
+                        [
+                            {"id": "devicenode", "config": GW_CONFIG_SAME_DEV_1}
+                        ])
+
+""" Following capabilities are fed to DynamicConfiguration tests to examine the case
+    when a new device is added to configuration
+"""
+test_cap_2 = Capability("double.activation.new.device.0.cap",
+                        [
+                            {"id": "devicenode", "config": GW_CONFIG_NEW_DEV_0}
+                        ])
+
+test_cap_3 = Capability("double.activation.new.device.1.cap",
+                        [
+                            {"id": "devicenode", "config": GW_CONFIG_NEW_DEV_1}
+                        ])
+
+""" Following capability is fed to DynamicConfiguration tests to examine the case
+    when exactly same device with same mode is added
+"""
+test_cap_4 = Capability("double.activation.same.device.same.mode.cap",
+                        [
+                            {"id": "devicenode", "config": GW_CONFIG_SAME_DEV_SAME_MODE}
+                        ])
+
+""" Define the manifest so it can be used by the testframework  """
+manifest = StandardManifest(TESTOUTPUT_DIR,
+                            "devicenode-test-manifest.json",
+                            [
+                                test_cap_0,
+                                test_cap_1,
+                                test_cap_2,
+                                test_cap_3,
+                                test_cap_4
+                            ])
+
+
+def service_manifests():
+    """ The agent fixture calls this function when it creates the service manifests
+        that should be used with this test module. The agent fixture expects a list
+        of StandardManifest and/or DefaultManifest objects.
+    """
+    return [manifest]
+
+# Test suites
+
+
+@pytest.mark.usefixtures("testhelper", "agent")
+class TestDeviceNode(object):
+    """ This suite tests the behavior of DeviceNodeGateway within given set of capabilities.
+        It uses the DeviceNodeGateway to assert behavior.
+
+        The tests use the Agent D-Bus interface to enable gateway configurations,
+        for a specified list of capabilities.
+    """
+
+    @pytest.mark.parametrize(("firstcap", "secondcap"), [
```

Comment:
It's difficult to understand how these combinations are meant to interact. In this case I'm guessing none of them should cause an exception and that's it? Otherwise, it might be a good idea to structure this so it's possible for the reader to understand "capA and capB in that order should mean that..."